dpppt-backend-sdk/dpppt-backend-sdk-data/src/test/java/org/dpppt/backend/sdk/data/gaen/GaenDataServiceTest.java

```diff
@@ -61,9 +64,51 @@ public void upsert() throws Exception {
 
     var returnedKeys =
         gaenDataService.getSortedExposedForKeyDate(
-            UTCInstant.today().minusDays(1), null, publishedUntil);
+            UTCInstant.today().minusDays(1), null, publishedUntil, now);
 
     assertEquals(keys.size(), returnedKeys.size());
     assertEquals(keys.get(1).getKeyData(), returnedKeys.get(0).getKeyData());
   }
+
+  @Test
+  public void testNoEarlyRelease() throws Exception {
+    Clock twoOClock = Clock.fixed(UTCInstant.today().plusHours(2).getInstant(), ZoneOffset.UTC);
+    Clock elevenOClock = Clock.fixed(UTCInstant.today().plusHours(11).getInstant(), ZoneOffset.UTC);
```

Comment:
I'm not very happy with the use of `setClock` in the tests - I would even remove it from `UTCInstant`. Even though I'm not sure if you could still run all the tests...

Solving this and solving a very improbable race-error (`UTCInstant.today()` _might_ change during the test), could give:

start with 
```
var today = UTCInstant.today()
```

use that throughout the code, and then only update `now` and pass that to the methods. 

Would that work? Or are there other `UTCInstant.now()` and `UTCInstant.today()` in the code-path?

Reply:
I've discussed it with @martinalig  but since it is very improbable we did not really care. It would only be a problem, iff the test is executed exactly at 00:00... but I can change it :)